buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RestTestRunnerTask.java

```diff
@@ -47,4 +57,19 @@ public int getMaxParallelForks() {
         return clusters;
     }
 
+    @Override
+    @Internal
+    public List<ResourceLock> getSharedResources() {
```

Comment:
I thought about that, as you say, for now it's only used by `Run` which will never be run in parallel with another similar task. Because we are overriding and internal method and using all sorts of other nasty internals here I wanted to limit the surface area of this until Gradle supports tasks declaring they use more than 1 lease of a build service.